plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts

```diff
@@ -31,11 +42,23 @@ export class AzureIdentityKubernetesAuthTranslator
       clusterDetails,
     );
 
-    const credentials = new DefaultAzureCredential();
+    if (!this.accessToken || this.tokenExpired()) {
+      this.accessToken = await this.tokenCredential.getToken(aksScope);
+
+      if (!this.accessToken) {
+        throw new Error('Unable to retrieve Azure token');
```

Comment:
Won't ever really reach this line right? I'm assuming the only way that `getToken` could fail is if it throws.

Reply:
I don't think it will, but the TokenCredential interface returns `string | null`, so this is just to protect against a potential null return.